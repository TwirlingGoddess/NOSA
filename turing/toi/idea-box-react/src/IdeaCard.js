import React from 'react';
import IdeaForm from './IdeaForm';

const IdeaCard = ({title, body, id, removeIdea}) => {
	return(
		<div className="Card">
			<h3>{title}</h3>
			<p>{body}</p>
			<button onClick={() => removeIdea(id)}>Delete</button>
		</div>
		)
}

export default IdeaCard