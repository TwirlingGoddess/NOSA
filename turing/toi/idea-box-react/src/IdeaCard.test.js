import React from 'react';
import { shallow } from 'enzyme';
import IdeaCard from './IdeaCard.js';

describe('IdeaCard', () => {
		const mockIdea = { title: 'Testing2', body: 'Now that I love it, I am unstoppable.', id: Date.now() }
	// beforeEach(() => {
	// })

	it('should have a title, body, and delete button', () => {
		let wrapper = shallow(<IdeaCard {...mockIdea}
								key={mockIdea.id}
								removeIdea={jest.fn()}
								/>)
		expect(wrapper.find('Card')).children.toBe(3)
	})

	it('should match the snapshot', () => {
		let wrapper = shallow(<IdeaCard {...mockIdea}
								key={mockIdea.id}
								removeIdea={jest.fn()}
								/>)
		expect(wrapper).toMatchSnapshot()
	})

	it('should invoke the removeIdea function when the delete button is clicked', () => {
		const mockEvent = jest.fn()
		let wrapper = shallow(<IdeaCard {...mockIdea}
								key={mockIdea.id}
								removeIdea={mockEvent}
								/>)
		wrapper.find('Card').simulate('click')
		expect(mockEvent).toHaveBeenCalled();
		expect(mockEvent).toHaveBeenCalledWith(mockIdea.id);

	})

})