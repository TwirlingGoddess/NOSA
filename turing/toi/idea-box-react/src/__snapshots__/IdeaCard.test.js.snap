// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCard
    body="Now that I love it, I am unstoppable."
    id={1534387276919}
    removeIdea={[Function]}
    title="Testing2"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Testing2
</h3>,
        <p>
          Now that I love it, I am unstoppable.
</p>,
        <button
          onClick={[Function]}
>
          Delete
</button>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Testing2",
        },
        "ref": null,
        "rendered": "Testing2",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Now that I love it, I am unstoppable.",
        },
        "ref": null,
        "rendered": "Now that I love it, I am unstoppable.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Testing2
</h3>,
          <p>
            Now that I love it, I am unstoppable.
</p>,
          <button
            onClick={[Function]}
>
            Delete
</button>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Testing2",
          },
          "ref": null,
          "rendered": "Testing2",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Now that I love it, I am unstoppable.",
          },
          "ref": null,
          "rendered": "Now that I love it, I am unstoppable.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
